src/blob_gc_job.cc

```diff
@@ -531,11 +531,14 @@ Status BlobGCJob::DeleteInputBlobFiles() {
     metrics_.gc_num_files++;
     RecordInHistogram(statistics(stats_), TITAN_GC_INPUT_FILE_SIZE,
                       file->file_size());
+    if (file->is_obsolete()) {
```

Comment:
`is_obsolete()` is not atomic since `BlobFileMeta::state_` is not atomic. Make `state_` an atomic?

Reply:
It seems that `BlobGCJob::DeleteInputBlobFiles()` call is guarded by `db_mutex_`. But not sure if other places that can change `BlobFileMeta::state_` are all guarded by `db_mutex_`. Making `state_` atomic is probably the safest.